LEVIN_PROTOCOL.md

```diff
@@ -0,0 +1,168 @@
+# Levin Protocol
+This is a document explaining the current design of the levin protocol, as
+used by Monero. The protocol is largely inherited from cryptonote, but has
+undergone some changes.
+
+This document also may differ from the `struct bucket_head2` in Monero's
+code slightly - the spec here is slightly more strict to allow for
+extensibility.
+
+One of the goals of this document is to clearly indicate what is being sent
+"on the wire" to identify metadata that could de-anonymize users over I2P/Tor.
+These issues will be addressed as they are found. See `ANONMITY_NETWORKS.md` in
+the top-level folder for any outstanding issues.
+
+> This document does not currently list all data being sent by the monero
+> protocol, that portion is a work-in-progress. Please take the time to do it
+> if interested in learning about Monero p2p traffic!
+
+
+## Header
+This header is sent for every Monero p2p message.
+
+>  0                   1                   2                   3
```

Comment:
Formatting looks off. Recommend formatting as a code block as opposed to quote.

Reply:
Looks great in vim! Will change to code block.

Reply:
Everything look great in vim ;)

Reply:
Note you might want to double check this whole table (e.g. looks like 10 bits per byte here and a couple of the field widths look off, such as command is 4 bytes not 2).

Reply:
Changed to graphical table to 8-bits (that was embarrassing! whoops!).

The `Command` field was intentionally listed as 2-bytes and the `Version` field was intentionally listed as 1-byte despite both being 4-bytes in the code. The fields are little-endian with the upper portions unused, so I marked them as reserved. I think that was inappropriate because there isn't a way to use them in a backwards compatible way. An older p2p parser would be interpreting things differently.

Both are being changed shortly.